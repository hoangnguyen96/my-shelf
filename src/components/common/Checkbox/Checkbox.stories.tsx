import { Meta, StoryFn } from "@storybook/react";
import { ChakraProvider } from "@chakra-ui/react";
import Checkbox from ".";
import theme from "@app/themes";

const meta: Meta<typeof Checkbox> = {
  component: Checkbox,
  decorators: [
    (Story: StoryFn) => (
      <ChakraProvider theme={theme}>
        <Story />
      </ChakraProvider>
    ),
  ],
};

export default meta;

const Template: StoryFn<typeof Checkbox> = (args) => <Checkbox {...args} />;

export const Default = Template.bind({});
Default.args = {
  text: "Remember me",
};
