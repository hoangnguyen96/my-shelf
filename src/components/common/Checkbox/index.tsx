import { Checkbox, CheckboxProps } from "@chakra-ui/react";

interface CheckboxBaseProps extends CheckboxProps {
  text?: string;
}

const CheckboxBase = ({ text = "Remember me", ...rest }: CheckboxBaseProps) => {
  return (
    <Checkbox size="md" {...rest}>
      {text}
    </Checkbox>
  );
};

export default CheckboxBase;
