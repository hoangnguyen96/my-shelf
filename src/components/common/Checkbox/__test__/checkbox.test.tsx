import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Checkbox from "..";

describe("Checkbox", () => {
  it("Should render correctly snapshot", () => {
    expect(render(<Checkbox />)).toMatchSnapshot();
  });
});
