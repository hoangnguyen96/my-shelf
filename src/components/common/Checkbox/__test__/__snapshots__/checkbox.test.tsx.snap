// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Should render correctly snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="chakra-checkbox css-1elxtiq"
      >
        <input
          class="chakra-checkbox__input"
          style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
          type="checkbox"
          value=""
        />
        <span
          aria-hidden="true"
          class="chakra-checkbox__control css-dnty2r"
        />
        <span
          class="chakra-checkbox__label css-1oeb2oe"
        >
          Remember me
        </span>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="chakra-checkbox css-1elxtiq"
    >
      <input
        class="chakra-checkbox__input"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="checkbox"
        value=""
      />
      <span
        aria-hidden="true"
        class="chakra-checkbox__control css-dnty2r"
      />
      <span
        class="chakra-checkbox__label css-1oeb2oe"
      >
        Remember me
      </span>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
